### PR TITLE
expanded code of conduct definitions and process

### PR DESCRIPTION
**Pull request by amoore877**
Evolve our process more and make escalation paths and definitions more clear.

### CODE REVIEW

File: CODE-OF-CONDUCT.md [hunk 1]
```diff
@@ -14,6 +16,40 @@ We follow the [CNCF Contributor Code of Conduct](https://github.com/cncf/foundat
...
+
+Green:
+In this interaction everyone was socially aware and respectful. Everyone considered and listened to others' perspectives, even if they disagreed with opinions or proposals.
+
+Yellow:
+In this interaction someone was overly reactive. A "knee jerk" reaction to a statement, outcome, or situation caused frustration, anger, or some other negativity to surface *and* result in impacting others' comfortability, feeling of belonging in the community, or otherwise caused disruption. It is completely human to have negative emotions, but we must respectfully manage our emotions and not unleash them upon others.
+Much like subtle-isms, nearly everyone will on occassion dip into Yellow. To err is human. It shows emotional maturity to recognize when this has happened and apologize for the behavior. It is a community concern if the situation keeps repeating, resulting in a warning and eventual banning.
+
+Red:
+Either frequent repeating of Yellow, or outright harassment, insults, etc. as listed non-exhaustively in Principles.
```
Comment (kfox1111) [inline CODE-OF-CONDUCT.md]: Maybe excessive rather then frequent?
```suggestion
Either excessive Yellows, or outright harassment, insults, etc. as listed non-exhaustively in Principles.
```

File: CODE-OF-CONDUCT.md [hunk 2]
```diff
@@ -14,6 +16,40 @@ We follow the [CNCF Contributor Code of Conduct](https://github.com/cncf/foundat
...
+These interactions contribute to a toxic environment, are disruptive to productivity, and cause an unwelcome and unsafe atmosphere.
```
Comment (kfox1111) [inline CODE-OF-CONDUCT.md]:
```suggestion
These interactions contribute to a toxic environment, are disruptive to productivity, and cause an unwelcoming or unsafe atmosphere.
```

File: CODE-OF-CONDUCT.md [hunk 3]
```diff
@@ -14,6 +16,40 @@ We follow the [CNCF Contributor Code of Conduct](https://github.com/cncf/foundat
...
+
+Retaliation against Code of Conduct reporters will result in a permanent ban.
+
+The SSC confidentially tracks all reports centrally, discussing information only with other CoC bodies under CNCF/Linux Foundation and only on an as-needed basis. All past SSC members are expected to continue to respect this confidentiality indefinitely- the SSC will consider it an egregious Code of Conduct violation to expose such information.
+
+When discussing violations with a user, the SSC will make reasonable attempts to generalize information so that it is still useful but not identifying of the reporter. This will not always be possible (eg., if a report is made regarding a 1-on-1 interaction).
+
+The SSC will issue a warning to users who don't follow this code of conduct. A second offense results in a temporary ban. A third offense warrants a permanent ban. It is at the SSC's discretion to un-ban offending users, or to immediately ban a toxic user without warning.
```
Comment (kfox1111) [inline CODE-OF-CONDUCT.md]: I get the desire to be able to unilaterally ban a toxic user without warning... but that may go a little too far except in the most extreme case (maybe just the violence use case listed above). We may not want to reserve the right unilaterally without bounds. That was a concern on the python article.

**Review by kfox1111 — COMMENTED** [review]
Thanks for working on this. Some suggestions/conversation inline.

File: CODE-OF-CONDUCT.md (hunk 3 above)
Comment (amoore877) [reply CODE-OF-CONDUCT.md]: updated

**Review by kfox1111 — APPROVED** [review]
LGTM. Thanks! :)

**Pull request merged**